astroquery/esa/xmm_newton/core.py

```diff
@@ -297,9 +301,146 @@ def _parse_filename(self, filename):
+        instruments : array of strings, optional, default []
+            An array of strings indicating the desired instruments
+        path: string, optional
+            If set, extracts the EPIC images in the indicated path
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A dictionary with the full paths of the extracted EPIC sources
+        spectral products. The key is the instrument
+        Notes
+        -----
+        The filenames will contain the source number in hexadecimal,
+        as this is the convention used by the pipeline.
+        The structure and the content of the extracted compressed FITS files
+        are described in details in the Pipeline Products Description
+        [XMM-SOC-GEN-ICD-0024](https://xmm-tools.cosmos.esa.int/external/xmm_obs_info/odf/data/docs/XMM-SOC-GEN-ICD-0024.pdf).
+        """
+        _instrument = ["M1", "M2", "PN", "EP"]
+        _product_type = ["SRSPEC", "BGSPEC", "SRCARF"]
+        _path = ""
+        ret = {}
+        if instrument == []:
+            instrument = _instrument
+        else:
+            for i in instrument:
+                if i not in _instrument:
+                    log.warning("Invalid instrument %s" % i)
+                    instrument.remove(i)
+        if path != "" and os.path.exists(path):
+            _path = path
+        try:
+            with tarfile.open(filename, "r") as tar:
+                for i in tar.getmembers():
+                    paths = os.path.split(i.name)
+                    fname = paths[1]
+                    paths = os.path.split(paths[0])
+                    if paths[1] != "pps":
+                        continue
+                    fname_info = self._parse_filename(fname)
+                    if fname_info["X"] != "P":
+                        continue
+                    if not fname_info["I"] in instrument:
+                        continue
+                    if not fname_info["T"] in _product_type:
+                        continue
+                    if int(fname_info["X-"], 16) != source_number:
+                        continue
+                    tar.extract(i, _path)
+                    key = fname_info["I"]
+                    value = os.path.abspath(os.path.join(_path, i.name))
+                    if fname_info["T"] == "BGSPEC":
+                        key = fname_info["I"] + "_bkg"
+                    elif fname_info["T"] == "SRCARF":
+                        key = fname_info["I"] + "_arf"
+                    else:
+                        with fits.open(value) as hdul:
+                            for ext in hdul:
+                                if ext.name != "SPECTRUM":
+                                    continue
+                                rmf_fname = ext.header["RESPFILE"]
+                                if fname_info["I"] == "M1" or \
```

Comment:
no need for line breaks, and below, we let longer than 80char long lines, the limit in the linter is set to 120, so keep the logical blocks together if they fit within that limit.

Removing the breaks would improve overall readability.

Reply:
Done, thanks